packages/ui/src/working-groups/modals/CreateOpening/types.tsx

```diff
@@ -0,0 +1,109 @@
+import { EventRecord } from '@polkadot/types/interfaces/system'
+import BN from 'bn.js'
+import * as Yup from 'yup'
+
+import { QuestionValueProps } from '@/common/components/EditableInputList/EditableInputList'
+import { ModalWithDataCall } from '@/common/providers/modal/types'
+import { BNSchema, minContext, moreThanMixed } from '@/common/utils/validation'
+import { GroupIdName } from '@/working-groups/types'
+
+export interface OpeningModalData {
+  group?: GroupIdName
+}
+
+export type CreateOpeningModalCall = ModalWithDataCall<'CreateOpening', OpeningModalData>
+
+// TODO research runtime constraints
+export interface OpeningConditions {
+  group: GroupIdName
+  minStake: BN
+  hiringTarget: number
+}
+
+export interface TransactionContext extends OpeningConditions {
+  transactionEvents?: EventRecord[]
+  openingId?: number
+}
+
+export const OpeningSchema = Yup.object().shape({
+  group: Yup.number().optional(),
+  target: Yup.number().optional(),
+  applicationForm: Yup.object().shape({
+    questions: Yup.array()
+      .of(
+        Yup.object({
+          questionField: Yup.string().required('Question is required'),
+          shortValue: Yup.boolean(),
+        })
+      )
+      .min(1)
+      .required('Questions is required'),
+  }),
+  durationAndProcess: Yup.object().shape({
+    details: Yup.string().required('Details is required'),
+    isLimited: Yup.boolean(),
+    duration: Yup.number().when('isLimited', {
+      is: true,
+      then: Yup.number().required('Duration is required'),
+    }),
+  }),
+  stakingPolicyAndReward: Yup.object().shape({
+    stakingAmount: BNSchema.test(
+      minContext('Input must be at least ${min} for proposal to execute', 'leaderOpeningStake', true, 'execution')
```

Comment:
where is the min variable or constant for this?
In the CLI, the minimum stake for workers is defined as 333 JOY and minimum unstaking period as 288000 blocks.
And the reward/block should be able to be set as zero. Some workers only get paid in discretionary spending.